svchost/disco/disco.go

```diff
@@ -63,6 +63,7 @@ func NewWithCredentialsSource(credsSrc auth.CredentialsSource) *Disco {
 	return &Disco{
 		hostCache: make(map[svchost.Hostname]*Host),
 		credsSrc:  credsSrc,
+		Transport: httpTransport,
```

Comment:
I looked at the code paths and think it's safe to update this here. 

But if we do, we should remove lines 101-104 and 160-164 as those then no longer serve any purpose and may give the wrong idea about our intentions there.

Reply:
So I thought `101-104` and `160-164` could stay there as a safeguard in case the caller decides to construct `Disco` without using these factory functions, but if we don't expect anyone to do that, then I guess crash is a reasonable behaviour? 🤔 

I don't have too strong opinion about it - happy to remove these lines if you're ok with this.

Reply:
Ah, very good point! 😃 I will remove these then. 👍